File: backend/DTO/Response/OrderItemResponse.cs
using System.ComponentModel.DataAnnotations;

namespace backend.DTO.Response
{
    public class OrderItemResponse
    {
        [Required]
        public required int OrderItemId { get; set; }
        [Required]
        public required int Quantity { get; set; }
        [Required]
        public required decimal Price { get; set; }
        [Required]
        public required int OrderId { get; set; }
        [Required]
        public required int ProductId { get; set; }
    }

}

File: frontend_admin/lib/model/response/order_item_response.dart
import 'package:equatable/equatable.dart';

class OrderItemResponse extends Equatable {
  const OrderItemResponse({this.orderItemId, this.quantity, this.price, this.orderId, this.productId});
  factory OrderItemResponse.fromJson(Map<String, dynamic> json) {
    return OrderItemResponse(
      orderItemId: json['orderItemId'],
      quantity: json['quantity'],
      price: json['price']?.toDouble(),
      orderId: json['orderId'],
      productId: json['productId'],
    );
  }
  final int? orderItemId;
  final int? quantity;
  final double? price;
  final int? orderId;
  final int? productId;

  Map<String, dynamic> toJson() {
    return {
      'orderItemId': orderItemId,
      'quantity': quantity,
      'price': price,
      'orderId': orderId,
      'productId': productId,
    };
  }

  @override
  List<Object?> get props => [orderItemId, quantity, price, orderId, productId];
}
